lib/AST/ASTPrinter.cpp

```diff
@@ -4785,12 +4787,24 @@ void PrintAST::visitErrorExpr(ErrorExpr *expr) {
   Printer << "<error>";
 }
 
-void PrintAST::visitTernaryExpr(TernaryExpr *expr) {}
+void PrintAST::visitTernaryExpr(TernaryExpr *expr) {
+  visit(expr->getCondExpr());
+  Printer << " ? ";
+  visit(expr->getThenExpr());
+  Printer << " : ";
+  visit(expr->getElseExpr());
+}
 
 void PrintAST::visitIsExpr(IsExpr *expr) {
+  if (expr->getType()) {
+    Printer << "true";
+  } else {
+    Printer << "false";
+  }
```

Comment:
I think there may be some misunderstanding here, to clarify; we're not trying to do any evaluation here, i.e we're not looking to produce a boolean result for an `IsExpr`. We are just printing the code, so this should look like `<expr> is <Type>`.